Deployment/Jenkins/master/Master-Elbv1.groovy

```diff
@@ -0,0 +1,138 @@
+    string(name: 'TemplateUrl', description: 'S3-hosted URL for the EC2 template file')
+    string(name: 'BackendTimeout', defaultValue: '600', description: 'How long - in seconds - back-end connection may be idle before attempting session-cleanup')
+    string(name: 'JenkinsAgentPort', description: 'TCP Port number that the Jenkins agent-hosts connect through')
+    string(name: 'JenkinsListenerCert', defaultValue: '', description: 'Name/ID of the ACM-managed SSL Certificate to protect public listener')
+    string(name: 'JenkinsListenPort', defaultValue: '443', description: 'TCP Port number on which the Jenkins ELB listens for requests')
+    string(name: 'JenkinsServicePort', defaultValue: '8080', description: 'TCP Port number that the Jenkins host listens to')
+    string(name: 'JenkinsPassesSsh', defaultValue: 'false', description: 'Whether to allow SSH passthrough to Jenkins master')
+    string(name: 'HaSubnets', description: 'Select three subnets - each from different Availability Zones')
+    string(name: 'ProxyPrettyName', description: 'A short, human-friendly label to assign to the ELB - no capital letters')
+    string(name: 'SecurityGroupIds', description: 'List of security groups to apply to the ELB')
+
+}
+
+stages {
+    stage ('Prepare Instance Environment') {
+        steps {
+            deleteDir()
+            git branch: "${GitProjBranch}",
+                credentialsId: "${GitCred}",
+                url: "${GitProjUrl}"
+            writeFile file: 'master.ec2.instance.parms.json',
+                text: /
+                [
+                {
+                  "ParameterKey": "BackendTimeout",
+                  "ParameterValue": "${env.BackendTimeout}"
+                },
+                {
+                  "ParameterKey": "JenkinsAgentPort",
+                  "ParameterValue": "${env.JenkinsAgentPort}"
+                },
+                {
+                  "ParameterKey": "JenkinsListenerCert",
+                  "ParameterValue": "${env.JenkinsListenerCert}"
+                },
+                {
+                  "ParameterKey": "JenkinsListenPort",
+                  "ParameterValue": "${env.JenkinsListenPort}"
+                },
+                {
+                  "ParameterKey": "JenkinsServicePort",
+                  "ParameterValue": "${env.JenkinsServicePort}"
+                },
+                {
+                  "ParameterKey": "JenkinsPassesSsh",
+                  "ParameterValue": "${env.JenkinsPassesSsh}"
+                },
+                {
+                  "ParameterKey": "HaSubnets",
+                  "ParameterValue": "${env.HaSubnets}"
+                },
+                {
+                  "ParameterKey": "ProxyPrettyName",
+                  "ParameterValue": "${env.ProxyPrettyName}"
+                },
+                {
+                  "ParameterKey": "SecurityGroupIds",
+                  "ParameterValue": "${env.SecurityGroupIds}"
+                }
```

Comment:
Continuation on prior nitpick &mdash; probably want this block to look more like:

~~~
                       }
                   ]
               /
~~~

Reply:
done